pkg/images/images.go

```diff
@@ -62,15 +63,26 @@ func (in Image) String() string {
 // the image tag is "latest", Always is returned; otherwise, the default pull policy is returned.
 // Other components are computed as follows: if the image specifies a (non-empty) component, that component is returned;
 // otherwise, the component from the default image is returned.
-func (in *Image) ApplyDefaults(defaults Image) *Image {
-	return &Image{
-		Registry:      in.registry(defaults),
-		Repository:    in.repository(defaults),
-		Name:          in.name(defaults),
-		Tag:           in.tag(defaults),
-		PullPolicy:    in.pullPolicy(defaults),
-		PullSecretRef: in.pullSecretRef(defaults),
+func (in *Image) Merge(i Image) *Image {
+	merged := in.DeepCopy()
+	if merged == nil {
+		merged = &Image{}
 	}
+	_ = mergo.Merge(merged, i)
+	if merged.Registry == "" {
```

Comment:
I wanted to do exactly that at the beginning, but we have a problem with step 2: the pull policy. Indeed the rule is: if the tag is `latest`, then the default pull policy is `Always`, otherwise, it's `IfNotPresent`. I can't express that condition with a single, default `Image` object.

Reply:
I understand your point. A few things here:

1. It is never necessary to explicitly set `latest` as a default; if a tag is unspecified, k8s itself defaults to using latest. Is there any reason we need to replicate that logic ourselves?
2. The `imagePullPolicy`  logic we are using here (`Always` if tag is `latest`, otherwise `IfNotPresent`) is again the default k8s behaviour (see [here](https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting)).

Does this mean that we can remove this logic given it is baked into k8s anyway? This would also allow us to keep the behaviour of the merge functions consistent.

Reply:
I understand your point. A few things here:

1. It is never necessary to explicitly set latest as a default; if a tag is unspecified, k8s itself defaults to using latest. Is there any reason we need to replicate that logic ourselves?
2. The imagePullPolicy logic we are using here (Always if tag is latest, otherwise IfNotPresent) is again the default k8s behaviour (see here).

Does this mean that we can remove this logic given it is baked into k8s anyway? This would also allow us to keep the behaviour of the merge functions consistent.

Reply:
Absolutely, I think you are right. I created an issue for it: https://github.com/k8ssandra/k8ssandra-operator/issues/532.

Reply:
Let's resolve this comment then.